fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/impl/content/registry/Taggable2ObjectMapRegistryImpl.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.content.registry;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.resource.ResourceType;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.content.registry.v1.util.Taggable2ObjectMap;
+import net.fabricmc.fabric.api.resource.ResourceManagerHelper;
+import net.fabricmc.fabric.api.resource.ResourceReloadListenerKeys;
+import net.fabricmc.fabric.api.resource.SimpleSynchronousResourceReloadListener;
+
+public class Taggable2ObjectMapRegistryImpl<K, V> implements Taggable2ObjectMap<K, V>, SimpleSynchronousResourceReloadListener {
```

Comment:
Really not sure whats going on with theses custom maps, seems super unnecessary

Reply:
Imo this should really keep a separate `Map<Identifier, V>` representing reload-persistent tag to value mappings, than doing all the hacks with runnables etc.

Reply:
That is not possible as it will not preserve the order in which things are added to the registry.

For example, if you remove BlockTags.LOGS from it and then add an entry with the key Blocks.OAK_LOG then you would want the OAK_LOG to be added *after* LOGS has been removed.